### PR TITLE
Add info to application and device payload formatters

### LINKED ISSUE #1566: Add info notification to the application payload formatters
#### Summary
The views for application payload formatters should have an info notification telling the user about the implications of setting application level payload formatters.

#### Why do we need this?
Improve clarity.

#### What is already there? What do you see now?
Application level payload formatter views.

#### What is missing? What do you want to see?
A notification at the top telling the user that application level payload formatters are a fallback setting for all devices of the application and that device level payload formatters always take precedence. @htdvisser please check

#### Environment
Browser.

#### How do you propose to implement this?
Trivial.

#### Can you do this yourself and submit a Pull Request?
@pgalic96

### PR DESCRIPTION
**Pull request by pgalic96**
#### Summary

Closes #1566 

#### Changes

- Add `<Notification info />` to application and payload formatter views

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, database and configuration, according to the stability commitments in `README.md`.
- [ ] Testing: The changes are covered with unit tests. The changes are tested manually as well.
- [ ] Documentation: Relevant documentation is added or updated.
- [ ] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

File: pkg/webui/console/views/application-payload-formatters/index.js [hunk 1]
```diff
@@ -108,10 +110,15 @@ export default class ApplicationPayloadFormatters extends React.Component {
       />
     )
 
+    const applicationFormatterInfo = (
+      <Notification className={style.notification} info content={m.infoText} />
```
Comment (kschiffer) [inline pkg/webui/console/views/application-payload-formatters/index.js]:
```suggestion
      <Notification className={style.notification} info content={m.infoText} small />
```
Please set the `small` flag for the link info notification as well.

File: pkg/webui/console/views/application-payload-formatters/index.js [hunk 2]
```diff
@@ -41,6 +41,8 @@ import {
 import style from './application-payload-formatters.styl'
 
 const m = defineMessages({
+  infoText:
+    'Application payload formatters are executed on all devices in the application, unless the user defines a device-specific payload formatter.',
```
Comment (htdvisser) [inline pkg/webui/console/views/application-payload-formatters/index.js]: These payload formatters are executed on uplink messages from all devices in this application. If payload formatters are also set for a specific device, only those will be executed for uplinks from that device.

File: pkg/webui/console/views/device-payload-formatters/index.js [hunk 3]
```diff
@@ -35,6 +37,9 @@ import { selectSelectedDeviceId } from '../../store/selectors/device'
 
 import style from './device-payload-formatters.styl'
 
+const m = defineMessages({
+  infoText: 'Device payload formatter overrides application payload formatters, if set.',
```
Comment (htdvisser) [inline pkg/webui/console/views/device-payload-formatters/index.js]: These payload formatters are executed on uplink messages from this device. If payload formatters are also set for the application, those are not executed for uplinks from this device.

**Comment by pgalic96** [comment]
@kschiffer since you added page titles in uplink and downlink components, adding specific messages introduces some repeated code so if you think there is a better way to pass these message in a `application-payload-decoder` let me know. same for linking warning which I think we should exclude from this PR, and fix in another.

Also I added a redirect on application payload formatter route which fixes the bug with route not found in application payload formatter.

File: pkg/webui/console/containers/application-payload-formatters/downlink.js [hunk 4]
```diff
@@ -91,9 +97,13 @@ class ApplicationPayloadFormatters extends React.PureComponent {
...
     const { formatters, linked } = this.props
 
+    const applicationFormatterInfo = <Notification small info content={m.infoText} />
+
     return (
       <React.Fragment>
-        <PageTitle title={sharedMessages.payloadFormattersDownlink} />
+        <PageTitle title={sharedMessages.payloadFormattersDownlink}>
+          {linked && applicationFormatterInfo}
+        </PageTitle>
```
Comment (kschiffer) [inline pkg/webui/console/containers/application-payload-formatters/downlink.js]: It doesn't need to be a child:
```js
<PageTitle title={sharedMessages.payloadFormattersDownlink} />
{linked && applicationFormatterInfo}
```

File: pkg/webui/console/containers/application-payload-formatters/uplink.js [hunk 5]
```diff
@@ -91,9 +97,13 @@ class ApplicationPayloadFormatters extends React.PureComponent {
...
     const { formatters, linked } = this.props
 
+    const applicationFormatterInfo = <Notification small info content={m.infoText} />
+
     return (
       <React.Fragment>
-        <PageTitle title={sharedMessages.payloadFormattersUplink} />
+        <PageTitle title={sharedMessages.payloadFormattersUplink}>
+          {linked && applicationFormatterInfo}
+        </PageTitle>
```
Comment (kschiffer) [inline pkg/webui/console/containers/application-payload-formatters/uplink.js]: See above.

**Pull request merged**